moto/dynamodb/responses.py

```diff
@@ -1142,6 +1258,25 @@ def transact_write_items(self) -> str:
                 item_attrs = item["Put"]["Item"]
                 table = self.dynamodb_backend.get_table(item["Put"]["TableName"])
                 validate_put_has_empty_keys(item_attrs, table)
+
+                condition_expression = item["Put"].get("ConditionExpression")
```

Comment:
This part seems to be the same for every action. Can we extract that somehow, to keep it DRY?

I don't want to prescribe exactly how you should do it, but I was thinking something along the lines of this:
```
for action in ["Put", ..]:
    if details := item.get(action):
        condition_expression = details.get("ConditionExpression")
        ...
```